src/azure-cli-core/azure/cli/core/__init__.py

```diff
@@ -96,9 +108,33 @@ def show_version(self):
         if updates_available == -1:
             logger.warning('Unable to check if your CLI is up-to-date. Check your internet connection.')
         elif updates_available:
-            logger.warning('You have %i updates available. Consider updating your CLI installation. '
-                           'Instructions can be found at https://docs.microsoft.com/en-us/cli/azure/install-azure-cli',
-                           updates_available)
+            warning_msg = 'You have %i updates available. Consider updating your CLI installation'
+            from azure.cli.core._environment import _ENV_AZ_INSTALLER
+            installer = os.getenv(_ENV_AZ_INSTALLER)
```

Comment:
do we really need to print different messages based on platform? what about show all upgrade instructions?

Reply:
The current [link](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli?view=azure-cli-latest) shown is doing something similiar, it puts all the links to each package in one page. But we got complains about it's not being useful, some users may not know their operating system well, they have difficulties to find the right instruction.